out/notebooks/spl2/initializer.ts

```diff
@@ -49,7 +48,7 @@ interface LSPLog {
 	message: string,
 }
 
-export async function startSpl2ClientAndServer(context: ExtensionContext, progressBar: StatusBarItem, portToAttempt: number, onClose: (nextPort: number) => void): Promise<Spl2ClientServer> {
+export async function startSpl2ClientAndServer(globalStoragePath: string, progressBar: StatusBarItem, portToAttempt: number, onClose: (nextPort: number) => void): Promise<Spl2ClientServer> {
```

Comment:
Factoring out the `context` object made testing possible, `context` is not easily mocked.

Reply:
Ohhh interesting, when I first tried this approach I used `vscode.extensions.getExtension("Splunk.splunk")` which didn't return anything useful, just some string metadata but I don't think I tried to call `activate` - let me give that a shot thanks!

Reply:
In either case I think it's a bit cleaner to refactor the code in this way though so that `context` can be limited to the `extension.js` code - that way the install/initialize code can be used independently

Reply:
ah unfortunately `activate()` is leading to its own set of issues: https://github.com/splunk/vscode-extension-splunk/actions/runs/5980833270/job/16227752726#step:14:35 I feel like we're making progress though, thanks for sharing that link